File: 剑指 Offer 54. 二叉搜索树的第k大节点/index2.js
/**
 * @deprecated 剑指 Offer 54. 二叉搜索树的第k大节点
 * @author 氧化氢
 */

/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} k
 * @return {number}
 */
 var kthLargest = function(root, k) {
  var arr = []
  getRoot(root, arr)
  
  return arr[arr.length - k]
};

var getRoot = function(root, arr) {
  if(!root) return
  getRoot(root.left, arr)
  arr.push(root.val)
  getRoot(root.right, arr)

  return
}

File: src/剑指 Offer 54. 二叉搜索树的第k大节点/index2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    """
    利用二叉搜索树得特性：左子树的值都比根节点小，右子树的值都比根节点大
    """

    def kthLargest(self, root: TreeNode, k: int) -> int:
        res = []
        self.mid_order(root, res)

        return res[len(res)-k]

    def mid_order(self, root, res):
        if root is None:
            return False
        self.mid_order(root.left, res)
        res.append(root.val)
        self.mid_order(root.right, res)
